src/components/ArticleCard.jsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticleById } from "../utils";
+import BeatLoader from "react-spinners/BeatLoader";
+
+export default function ArticleCard() {
+  const [article, setArticle] = useState({});
+  const { article_id } = useParams();
+  const [isLoading, setIsLoading] = useState(true);
+  let [color, setColor] = useState("#ffffff");
+
+  useEffect(() => {
+    fetchArticleById(article_id)
+      .then((res) => {
+        return res.result;
+      })
+      .then((article) => {
+        setArticle(article[0]);
+        setIsLoading(false);
+      });
+  }, []);
+
+  if (isLoading) {
+    return (
+      <>
+        <h3>waiting for your data to magically appear</h3>
+        <BeatLoader
+          color={color}
```

Comment:
Do we need color as state?

Reply:
If I don't set the colour in state it will just display the spinner as black. That's why I left it there as it was in the documentation.